impeller/entity/entity_pass.cc

```diff
@@ -585,9 +585,12 @@ EntityPass::EntityResult EntityPass::GetEntityForElement(
       auto texture = pass_context.GetTexture();
       // Render the backdrop texture before any of the pass elements.
       const auto& proc = subpass->backdrop_filter_proc_;
-      subpass_backdrop_filter_contents =
-          proc(FilterInput::Make(std::move(texture)),
-               subpass->transform_.Basis(), Entity::RenderingMode::kSubpass);
+
+      subpass_backdrop_filter_contents = proc(
+          FilterInput::Make(std::move(texture)), subpass->transform_.Basis(),
+          subpass->bounds_promise_ == ContentBoundsPromise::kMayClipContents
+              ? Entity::RenderingMode::kClippedSubpass
+              : Entity::RenderingMode::kSubpass);
```

Comment:
@flar I found experimentally that this is the appropriate check to determine the math in the `MatrixFilterContents`.  This is a bit odd since the verbiage includes "may".  While that isn't a definite clause, it seems to have a real change on the math.  I was hoping you could provide more light on this.  Thanks.